flake.nix

```diff
@@ -29,15 +29,15 @@
 
           libraryHaskellDepends = [
```

Comment:
Should we just separate this out into `moat.nix` or use `callCabal2Nix`?

Reply:
I think `callCabal2Nix` would be fine.

Reply:
Looks like `callCabal2nix` uses IFD, which breaks `nix flake check` and `nix flake show`: https://github.com/NixOS/nix/issues/4265

I'll try splitting it out into `moat.nix` for now.